lib/ansible/plugins/vars/host_group_vars.py

```diff
@@ -136,14 +136,20 @@ def get_vars(self, loader, path, entities, cache=True):
 
                     if os.path.isdir(opath):
                         self._display.debug("\tprocessing dir %s" % opath)
-                        FOUND[key] = found_files = loader.find_vars_files(opath, entity_name)
+                        if cache:
+                            key = '%s.%s' % (entity_name, opath)
+                            FOUND[key] = found_files = loader.find_vars_files(opath, entity_name)
+                        else:
+                            found_files = loader.find_vars_files(opath, entity_name)
```

Comment:
the real fix is moving key definition from 130 to 109, all this can then stay as it was

Reply:
If I move key definition from 130 to 109, it would have issues with opath not being found as its defined afterwards. 

Should I move opath as well?

Reply:
good catch, no, i thnk that then moving both to 121 makes more sense

Reply:
I've made the changes